src/io_threads.c

```diff
@@ -501,7 +503,12 @@ int tryOffloadFreeObjToIOThreads(robj *obj) {
         return C_ERR;
     }
 
-    IOJobQueue_push(jq, decrRefCountVoid, obj);
+    /* We offload only the free of the ptr that may be allocated by the I/O thread.
+     * The object itself was allocated by the main thread and will be freed by the main thread. */
+    IOJobQueue_push(jq, sdsfree, obj->ptr);
```

Comment:
Hm... The compiler doesn't like sdsfree here because it expects `void (*)(void *)`.

I guess we have two options:
1. Create a wrapper that takes a void *.
2. Cast it to `void (*)(void *)`, via `(void *)` to avoid the warnings.

```suggestion
    IOJobQueue_push(jq, (void (*)(void *))(void *)sdsfree, obj->ptr);
```

WARNING: Untested.

Reply:
:bulb: Better idea:

```suggestion
    IOJobQueue_push(jq, zfree, sdsAllocPtr(obj->ptr));
```

Reply:
This will require memory access to the ptr which we try to avoid.

Reply:
Maybe we can add sdsfreeVoid version as we did with decrRefCound for obj.
```
/* This variant of decrRefCount() gets its argument as void, and is useful
 * as free method in data structures that expect a 'void free_object(void*)'
 * prototype for the free method. */
void decrRefCountVoid(void *o) {
    decrRefCount(o);
}
```

Reply:
> This will require memory access to the ptr

Good point.

> sdsfreeVoid

Yes, this is option 1 in my first comment. You don't like `IOJobQueue_push(jq, (void (*)(void *))(void *)sdsfree, obj->ptr);`?  It's too ugly, right?

OK, probably sdsfreeVoid is better.

Reply:
can't we just typedef the (void (*)(void *))(void *) and cast to the type?

Reply:
Yes, that would the same, but maybe more readable. The `(void *)` in end is an extra cast just to avoid warnings for casting between incompatible type.

Reply:
Added `sdsfreeVoid` and removed `decrRefCountVoid` which is no longer in use.